packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/OutputLibraryTests.cs

```diff
@@ -26,9 +28,54 @@ public void BuildTypeProviders_Override()
             Assert.Throws<NotImplementedException>(() => { object shouldFail = _outputLibrary.TypeProviders; });
         }
 
-        internal class MockOutputLibrary : OutputLibrary
+        [Test]
+        public void CanAddVisitors()
+        {
+            _outputLibrary.AddVisitor(new MockOutputLibraryVisitor());
+            Assert.AreEqual(1, _outputLibrary.GetOutputLibraryVisitors().Count());
+        }
+
+        [Test]
+        public void CanOverrideGetOutputLibraryVisitors()
+        {
+            var outputLibrary = new MockOutputLibraryOverridingVisitors(new [] { new MockOutputLibraryVisitor() });
+            Assert.AreEqual(1, outputLibrary.GetOutputLibraryVisitors().Count());
+
+            outputLibrary.AddVisitor(new MockOutputLibraryVisitor());
+            Assert.AreEqual(2, outputLibrary.GetOutputLibraryVisitors().Count());
+        }
+
+        private class MockOutputLibraryVisitor : OutputLibraryVisitor
```

Comment:
nit: can we call this TestOutputLibraryVisitor so its not confused with a Moq visitor.